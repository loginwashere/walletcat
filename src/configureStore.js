import { createStore, applyMiddleware, compose } from 'redux'
import reducers from './reducers'
import middleware from './middleware'
import { addLocaleData } from 'react-intl'
import { DEFAULT_LOCALE } from './config'
import localeMessages from './translations'

import enLocaleData from 'react-intl/locale-data/en'
import ruLocaleData from 'react-intl/locale-data/ru'
import ukLocaleData from 'react-intl/locale-data/uk'

addLocaleData(enLocaleData)
addLocaleData(ruLocaleData)
addLocaleData(ukLocaleData)

const locale = DEFAULT_LOCALE
const messages = localeMessages[locale]

const initialState = {
  intl: {
    locale,
    messages,
  }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const configureStore = (preloadedState = {}) => createStore(
  reducers,
  {
    ...preloadedState,
    ...initialState
  },
  composeEnhancers(applyMiddleware(...middleware))
)

export default configureStore
