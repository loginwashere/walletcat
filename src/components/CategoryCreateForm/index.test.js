import React from 'react'
import { reduxForm } from 'redux-form'
import renderer from 'react-test-renderer'
import { CategoryCreateForm } from '.'
import configureStore from '../../configureStore'
import { Provider } from 'react-redux'

const Decorated = reduxForm({ form: 'testForm' })(CategoryCreateForm)

describe('components:CategoryCreateForm:', () => {
  it('renders without crashing', () => {
    const store = configureStore()
    const dispatch = jest.fn()
    const handleSubmit = jest.fn()
    const currentPage = 1
    const tree = renderer.create(
      <Provider store={store} >
        <Decorated handleSubmit={handleSubmit}
                  submitting={false}
                  pristine={true}
                  invalid={false}
                  reset={dispatch}
                  currentPage={currentPage} />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
