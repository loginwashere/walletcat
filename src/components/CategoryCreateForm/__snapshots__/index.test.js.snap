exports[`components:CategoryCreateForm: renders without crashing 1`] = `
<form
  className="form-horizontal"
  onSubmit={[Function]}>
  <div
    className="form-group has-feedback">
    <h1
      className="truncate">
      New Category
    </h1>
    <div />
  </div>
  <div>
    <div
      className="form-group has-feedback">
      <label
        className="col-sm-2 control-label"
        htmlFor="form-horizontal-name">
        Name
      </label>
      <div
        className="col-sm-10">
        <input
          autoFocus={undefined}
          className="form-control"
          id="form-horizontal-name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value="" />
      </div>
    </div>
    <div
      className="form-group has-feedback">
      <label
        className="col-sm-2 control-label"
        htmlFor="form-horizontal-description">
        Description
      </label>
      <div
        className="col-sm-10">
        <input
          autoFocus={undefined}
          className="form-control"
          id="form-horizontal-description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Description"
          required={undefined}
          type="text"
          value="" />
      </div>
    </div>
  </div>
  <div
    className="form-group">
    <div
      className="col-sm-2 col-sm-offset-2 col-xs-4">
      <button
        action="push"
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button">
        Cancel
      </button>
    </div>
    <div
      className="col-sm-2 col-xs-4">
      <button
        className="btn btn-default"
        disabled={false}
        type="submit">
        Create
      </button>
    </div>
    <div
      className="col-sm-2 col-xs-4">
      <button
        className="btn btn-default"
        disabled={true}
        onClick={[Function]}
        type="button">
        Clear
      </button>
    </div>
  </div>
</form>
`;
